pkg/application/registryConfigUseCase.go

```diff
@@ -0,0 +1,225 @@
+		blueprintRepository:           blueprintRepository,
+		doguConfigRepository:          doguConfigRepository,
+		doguSensitiveConfigRepository: doguSensitiveConfigRepository,
+		globalConfigRepository:        globalConfigRepository,
+	}
+}
+
+// ApplyConfig fetches the dogu and global config statediff of the blueprint and applies these keys to the repositories.
+func (useCase *EcosystemRegistryUseCase) ApplyConfig(ctx context.Context, blueprintId string) error {
+	logger := log.FromContext(ctx).WithName("EcosystemRegistryUseCase.ApplyConfig").
+		WithValues("blueprintId", blueprintId)
+
+	blueprintSpec, err := useCase.blueprintRepository.GetById(ctx, blueprintId)
+	if err != nil {
+		return fmt.Errorf("cannot load blueprint to apply config: %w", err)
+	}
+
+	// TODO Remove this before merge.
+	// stage := os.Getenv("STAGE")
+	// if stage == "development" {
+	// 	logger.Info("set config diffs from mock data...")
+	// 	data := parseDoguConfigDiffMockData()
+	// 	logger.Info("dogu config diffs:")
+	// 	logger.Info(fmt.Sprintf("%+v", data))
+	// 	blueprintSpec.StateDiff.DoguConfigDiff = data
+	// 	logger.Info("dogu config diffs in statediff:")
+	// 	logger.Info(fmt.Sprintf("%+v", blueprintSpec.StateDiff.DoguConfigDiff))
+	// 	blueprintSpec.StateDiff.GlobalConfigDiff = parseGlobalConfigDiffMockData()
+	// }
+
+	doguConfigDiffs := blueprintSpec.StateDiff.DoguConfigDiff
+	isEmptyDoguDiff := len(doguConfigDiffs) == 0
+	if isEmptyDoguDiff {
+		logger.Info("dogu config diffs are empty...")
+	}
+
+	globalConfigDiffs := blueprintSpec.StateDiff.GlobalConfigDiff
+	isEmptyGlobalDiff := len(globalConfigDiffs) == 0
+	if isEmptyGlobalDiff {
+		logger.Info("global config diffs are empty...")
+	}
+
+	if isEmptyDoguDiff && isEmptyGlobalDiff {
+		// TODO Correct Status or create new for no Action needed?
+		return useCase.markConfigApplied(ctx, blueprintSpec)
+	}
+
+	err = useCase.markApplyConfigStart(ctx, blueprintSpec)
+	if err != nil {
+		return useCase.handleFailedApplyRegistryConfig(ctx, blueprintSpec, err)
+	}
+
+	var errs []error
+	for doguName, doguDiffs := range doguConfigDiffs {
+		errs = append(errs, useCase.applyDoguConfigDiffs(ctx, doguName, doguDiffs.DoguConfigDiff))
+		errs = append(errs, useCase.applySensitiveDoguConfigDiffs(ctx, doguName, doguDiffs.SensitiveDoguConfigDiff))
+	}
+
+	errs = append(errs, useCase.applyGlobalConfigDiffs(ctx, globalConfigDiffs))
+
+	joinedErr := errors.Join(errs...)
+	if joinedErr != nil {
+		return useCase.handleFailedApplyRegistryConfig(ctx, blueprintSpec, joinedErr)
+	}
+
+	return useCase.markConfigApplied(ctx, blueprintSpec)
+}
+
+func parseGlobalConfigDiffMockData() domain.GlobalConfigDiff {
+	object := &domain.GlobalConfigDiff{}
+	err := yaml.Unmarshal(globalConfigDiffMockDataBytes, object)
+	if err != nil {
+		panic(fmt.Errorf("error during mock data deserialization"))
+	}
+
+	return *object
+}
+
+func parseDoguConfigDiffMockData() map[common.SimpleDoguName]domain.CombinedDoguConfigDiff {
```

Comment:
🚫 **[checkstyle]** <unused> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>func `parseDoguConfigDiffMockData` is unused